keel-core/src/main/kotlin/com/netflix/spinnaker/keel/api/Environment.kt

```diff
@@ -1,8 +1,36 @@
 package com.netflix.spinnaker.keel.api
 
+import com.fasterxml.jackson.annotation.JsonIgnore
+
 data class Environment(
   val name: String,
+  @JsonIgnore
+  var id: String? = null,
```

Comment:
Couldn't we default this to `randomUID().toString()` instead to avoid making it nullable? When loading from the repo it will be non-null, otherwise it seems we're assuming it's non-null everywhere anyway...

Reply:
That would break things. The way we're assigning uuids on objects such as `Environment` but hiding them from users makes it difficult to utilize them directly (in places where using a uuid directly is more efficient.)

Let me explore mapping `ConstraintState` to an environment via `DeliveryConfig.name` + `Environment.name` instead of always using environment.uuid, would enable reverting the addition of `id` on `Environment`.